### ISSUE TITLE
Move`_check_maps` and `_check_blocks` into public `equal` and `equal_block` functions

### ISSUE THREAD

**Issue by PicoCentauri** [lab-cosmo/equistore#119]
In #115 we discussed that the two functions `_check_maps` and `_check_blocks` which check if meta data between two TensorMaps/TensorBlocks is the same. Since @Luthaf has a valid point of not testing private function I propose we make them public but change their scope a bit.

The function should also have the option to check the values/gradients and not only the meta data. With this we can create a functions similar to `allclose`. I think of something like:

```python
def equal(
    tensor1: TensorMap,
    tensor2: TensorMap,
    test_data: bool=True,
) -> bool:
    ...
   
def equal_raise(
    tensor1: TensorMap,
    tensor2: TensorMap,
    test_data: bool=True,
) -> bool:
    ...

def equal_block(
    tensor1: TensorMap,
    tensor2: TensorMap,
    test_data: bool=True,
) -> bool:
    ...

def equal_block_raise(
    tensor1: TensorMap,
    tensor2: TensorMap,
    test_data: bool=True,
) -> bool:
    ...
```

**Comment by jwa7** [comment]
An argument of semantics / UX philosophy here, but should the out-of-the-box behaviour of a function like ``equal`` be to act on the whole TensorMap, i.e. data and metadata? I know you have default behaviour ``test_data=True`` here, but I think negating the argument to enforce explicit specification of metadata-only comparison makes more sense.

For instance change from:
```py
def equal(
    tensor1: TensorMap,
    tensor2: TensorMap,
    test_data: bool = True,
) -> bool:
    ...
```
to:
```py
def equal(
    tensor1: TensorMap,
    tensor2: TensorMap,
    metadata_only: bool = False,
) -> bool:
    ...
```

Such that:
```py
# Compare whole TensorMaps, i.e. an X and Y
equal(tensor1=X, tensor2=Y)
>>> False

# Compare just metadata, i.e. an X and Y
equal(tensor1=X, tensor2=Y, metadata_only=True)
>>> True
```

**Comment by jwa7** [comment]
But to add I think having these functions as public is a great idea, I find myself using them (or variations of them) a lot

**Comment by DavideTisi** [comment]
i close this since we merged #121 which implements what is requested here
